dev/plugins/hu.elte.txtuml.diagnostics/src/hu/elte/txtuml/diagnostics/animation/js/DiagnosticsServer.java

```diff
@@ -0,0 +1,146 @@
+
+	private HttpServer server;
+	private ConcurrentMap<String, RegistryEntry> registry = new ConcurrentHashMap<>();
+	private DiagnosticsPlugin diagnosticsPlugin;
+	
+	public DiagnosticsServer(DiagnosticsPlugin diagnosticsPlugin){
+		this.diagnosticsPlugin = diagnosticsPlugin;
+	}
+
+	public void start(Integer port) throws IOException {
+		server = HttpServer.create(new InetSocketAddress(port), 0);
+		server.createContext("/" + GlobalSettings.TXTUML_DIAGNOSTICS_HTTP_PATH, new DiagnosticsHandler());
+		server.createContext("/" + GlobalSettings.TXTUML_DIAGNOSTICS_DELAY_PATH, new DelayHandler());
+		server.createContext("/" + GlobalSettings.TXTUML_DIAGNOSTICS_DELAY_INPUT_PATH, new DelayInputHandler());
+		server.setExecutor(null); // creates a default executor
+		server.start();
+	}
+
+	public void stop() {
+		if (server != null) server.stop(0);
+		registry.clear();
+	}
+
+	public void register(ModelEvent event) {
+		RegistryEntry registryEntry = new RegistryEntry(event.modelClassName,
+				event.modelClassInstanceName, event.eventTargetClassName);
+		registry.put(event.modelClassInstanceID, registryEntry);
+	}
+
+	private class DiagnosticsHandler implements HttpHandler {
+
+		@Override
+		public void handle(HttpExchange exchange) throws IOException {
+			// Set required headers
+			Headers headers = exchange.getResponseHeaders();
+			headers.add("Content-type", "application/json");
+			headers.add("Access-Control-Allow-Origin", "*");
+
+			// Build the payload
+			String response = registry.entrySet().stream()
+					.map(DiagnosticsServer::registryEntryToJson)
+					.collect(Collectors.joining(",", "[", "]")); 
+
+			// Write response
+			exchange.sendResponseHeaders(200, response.length());
+			OutputStream os = exchange.getResponseBody();
+			os.write(response.getBytes());
+			os.close();
+		}
+
+	}
+	
+	private class DelayHandler implements HttpHandler {
+
+		@Override
+		public void handle(HttpExchange exchange) throws IOException {
+			//Get request data
+			String body = "";
+		    try (InputStreamReader reader = new InputStreamReader(exchange.getRequestBody())) {
+				body = new BufferedReader(reader).lines().collect(Collectors.joining("\n"));
+		    }
+		    
+            //Get and set animation delay
+		    String[] delayArray = body.split("=");
+		    int delay = Integer.parseInt(delayArray[1]);
+			setAnimationDelay(delay);
+			
+			// Write response
+			exchange.sendResponseHeaders(200, 0);
+			OutputStream os = exchange.getResponseBody();
+			os.close();
+		}
+
+	}
+	
+	private class DelayInputHandler implements HttpHandler {
+
+		@Override
+		public void handle(HttpExchange exchange) throws IOException {
+			// Set required headers
+			Headers headers = exchange.getResponseHeaders();
+			headers.add("Content-type", "application/json");
+			headers.add("Access-Control-Allow-Origin", "*");
+			
+			// Build the payload
+			String response = "["+ "{\"delayTime\":\"" + (diagnosticsPlugin.getDelay()/1000) + "\"}" + "]";
```

Comment:
I recommend sending the delay as a single value instead of a one-element list.

Reply:
While I think we should keep to a JSON format, as in key-value pairs, encapsulating it in an array is unnecessary.